### PR TITLE
Dynamic certificates loading

### PR DESCRIPTION
**Pull request by TomerYakir**
Manually tested that when passing additional certificates, the correct certificate is shown according to SSLServerName.
Also tested backwards compatibility with existing application

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -10,14 +10,67 @@ import "time"
...
+}
+
+func (s *SyncTlsConfig) getTlsConfig() *tls.Config {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	return s.tlsConfig
+}
+
+func (s *SyncTlsConfig) setTlsConfig(sslKeys []*SSLPair, cipherSuites []uint16, minTlsVersion uint16, fallbackKeys []SSLPair) error {
+	s.lock.Lock()
```
Comment (tolsen) [inline server.go]: what do you think about moving the grabbing of the lock to just before we set s.tlsConfig?

File: server.go [hunk 2]
```diff
@@ -10,14 +10,67 @@ import "time"
...
 type ServerConfig struct {
 	BindHost string
 	BindPort int
 
 	UseSSL        bool
 	SSLKeys       []SSLPair
-	MinTlsVersion uint16 // see tls.Version* constants
+	SyncTlsConfig *SyncTlsConfig
 
+	MinTlsVersion      uint16
```
Comment (tolsen) [inline server.go]: lost a comment here?

File: server.go (hunk 1 above)
Comment (TomerYakir) [reply server.go]: good idea - fixed

File: server.go (hunk 2 above)
Comment (TomerYakir) [reply server.go]: readded

**Comment by TomerYakir** [comment]
This is partially backwards compatible through the "fallback keys" mechanism.
The existing Atlas/MMS code won't work as-is and would require some minor modifications to work, specifically calling `proxy.InitializeServer()` and `proxy.OnSSLConfig()` before invoking `proxy.Run()`.

**Pull request merged**